axelrod/tests/strategies/test_axelrod_second.py

```diff
@@ -694,3 +694,128 @@ def test_strategy(self):
+        self.versus_test(Defect1_37, expected_actions=actions, seed=2)
+
+        # However a defect on turn 38 would be considered a burn.
+        opponent_actions = [D] + [C] * 36 + [D] + [C] * 9
+        Defect1_38 = axelrod.MockPlayer(actions=opponent_actions)
+        # Tit-for-Tat on the first, but no streaks, no Fair-weather flag.
+        actions = [(C, D), (D, C)] + [(C, C)] * 34 + [(D, C)]
+        # Two cooperations scheduled after the 37-turn defection
+        actions += [(C, D), (C, C)]
+        # TFT from then on, since burned
+        actions += [(C, C)] * 8
+        self.versus_test(Defect1_38, expected_actions=actions, seed=2)
+
+        # Use alternator to test parity flags.
+        actions = [(C, C), (C, D)]
+        # Even streak is set to 2, one for the opponent's defect and one for
+        # our defect.
+        actions += [(D, C)]
+        actions += [(C, D)]
+        # Even streak is increased two more.
+        actions += [(D, C)]
+        actions += [(C, D)]
+        # Opponent's defect increments even streak to 5, so we cooperate.
+        actions += [(C, C)]
+        actions += [(C, D), (D, C), (C, D), (D, C), (C, D)]
+        # Another 5 streak
+        actions += [(C, C)]
+        # Repeat
+        actions += [(C, D), (D, C), (C, D), (D, C), (C, D), (C, C)] * 3
+        # Repeat.  Notice that the last turn is the 37th move, but we do not
+        # defect.
+        actions += [(C, D), (D, C), (C, D), (D, C), (C, D), (C, C)]
+        self.versus_test(axelrod.Alternator(), expected_actions=actions)
+
+        # Test for parity limit shortening.
+        opponent_actions = [D, C] * 1000
+        AsyncAlternator = axelrod.MockPlayer(actions=opponent_actions)
+        actions = [(C, D), (D, C), (C, D), (D, C), (C, D), (C, C)] * 6
+        # Defect on 37th move
+        actions += [(D, D)]
+        actions += [(C, C)]
+        # This triggers the burned flag.  We should just Tit-for-Tat from here.
+        actions += [(C, D)]
+        actions += [(D, C), (C, D), (D, C), (C, D), (C, C)]
+        # This is the seventh time we've hit the limit.  So do it once more.
+        actions += [(C, D), (D, C), (C, D), (D, C), (C, D), (C, C)]
+        # Now hit thi limit sooner
+        actions += [(C, D), (D, C), (C, D), (C, C)] * 5
+        self.versus_test(AsyncAlternator, expected_actions=actions)
+
+        # Use a Defector to test the 20-defect streak
+        actions = [(C, D), (D, D), (D, D), (D, D), (D, D)]
+        # Now the two parity flags are used
+        actions += [(C, D), (C, D)]
+        # Repeat
+        actions += [(D, D), (D, D), (D, D), (D, D), (C, D), (C, D)] * 2
+        actions += [(D, D)]
+        # 20 D have passed
+        actions += [(D, D)] * 100
+        # The defect streak will always be detected from here on, because it
+        # doesn't reset.  This logic comes before parity streaks or the turn-
+        # based logic.
+        self.versus_test(axelrod.Defector(), expected_actions=actions)
```

Comment:
Another thought, could we add a few `attrs={"mode": "Normal", "recorded_defects": ...}` to the `versus_test` call so that we specifically test some of these attribute values? Not necessarily overkill but at least the `mode` and perhaps one or two others as you see fit?

Reply:
That's cool.  Didn't know that was possible.